diff --git a/kernel/arch/arm64/configs/mido_defconfig b/kernel/arch/arm64/configs/mido_defconfig
--- a/kernel/arch/arm64/configs/mido_defconfig
+++ b/kernel/arch/arm64/configs/mido_defconfig
@@ -158,7 +158,7 @@ CONFIG_CPUSETS_ASSIST=y
 CONFIG_CPUSET_BG="0-1"
 CONFIG_CPUSET_AUDIO="0-3"
 CONFIG_CPUSET_CAMERA="0-7"
-CONFIG_CPUSET_FG="0-5"
+CONFIG_CPUSET_FG="0-3"
 CONFIG_CPUSET_RESTRICTED="0-3"
 CONFIG_CPUSET_SYSTEM_BG="0-3"
 CONFIG_CPUSET_TOP_APP="0-7"
diff --git a/kernel/arch/arm/boot/dts/qcom/msm8953.dtsi b/kernel/arch/arm/boot/dts/qcom/msm8953.dtsi
--- a/kernel/arch/arm/boot/dts/qcom/msm8953.dtsi
+++ b/kernel/arch/arm/boot/dts/qcom/msm8953.dtsi
@@ -23,7 +23,7 @@
 	interrupt-parent = <&intc>;
 
 	chosen {
-		bootargs = "core_ctl_disable_cpumask=0-7 sched_enable_hmp=1 sched_enable_power_aware=1";
+		bootargs = "sched_enable_power_aware=0";
 	};
 
 	firmware: firmware {
diff --git a/kernel/arch/arm/boot/dts/qcom/msm8953-regulator.dtsi b/kernel/arch/arm/boot/dts/qcom/msm8953-regulator.dtsi
--- a/kernel/arch/arm/boot/dts/qcom/msm8953-regulator.dtsi
+++ b/kernel/arch/arm/boot/dts/qcom/msm8953-regulator.dtsi
@@ -17,9 +17,9 @@
 	rpm-regulator-smpa1 {
 		status = "okay";
 		pm8953_s1: regulator-s1 {
-			regulator-min-microvolt = <845000>;
-			regulator-max-microvolt = <1131000>;
-			qcom,init-voltage = <975000>;
+			regulator-min-microvolt = <870000>;
+			regulator-max-microvolt = <1156000>;
+			qcom,init-voltage = <1000000>;
 			status = "okay";
 		};
 	};
diff --git a/kernel/fs/sync.c b/kernel/fs/sync.c
--- a/kernel/fs/sync.c
+++ b/kernel/fs/sync.c
@@ -18,7 +18,7 @@
 #include <linux/backing-dev.h>
 #include "internal.h"
 
-bool fsync_enabled = false;
+bool fsync_enabled = true;
 module_param(fsync_enabled, bool, 0755);
 
 #define VALID_FLAGS (SYNC_FILE_RANGE_WAIT_BEFORE|SYNC_FILE_RANGE_WRITE| \